dspace-api/src/main/java/org/dspace/authenticate/SamlAuthentication.java

```diff
@@ -0,0 +1,750 @@
+     *
+     * It is safe to call this method multiple times.
+     *
+     * @param context context
+     * @throws SQLException if database error
+     */
+    protected synchronized void initialize(Context context) throws SQLException {
+        if (metadataHeaderMap != null) {
+            return;
+        }
+
+        HashMap<String, String> map = new HashMap<>();
+
+        String[] mappingString = configurationService.getArrayProperty("authentication-saml.eperson.metadata");
+
+        boolean autoCreate = configurationService
+            .getBooleanProperty("authentication-saml.eperson.metadata.autocreate", true);
+
+        // Bail out if not set, returning an empty map.
+
+        if (mappingString == null || mappingString.length == 0) {
+            log.debug("No additional eperson metadata mapping found: authentication-saml.eperson.metadata");
+
+            metadataHeaderMap = map;
+            return;
+        }
+
+        log.debug("Loading additional eperson metadata from: authentication-saml.eperson.metadata="
+            + StringUtils.join(mappingString, ","));
+
+        for (String metadataString : mappingString) {
+            metadataString = metadataString.trim();
+
+            String[] metadataParts = metadataString.split("=>");
+
+            if (metadataParts.length != 2) {
+                log.error("Unable to parse metadata mapping string: '" + metadataString + "'");
+
+                continue;
+            }
+
+            String attributeName = metadataParts[0].trim();
+            String metadataFieldName = metadataParts[1].trim().toLowerCase();
+
+            boolean valid = checkIfEpersonMetadataFieldExists(context, metadataFieldName);
+
+            if (!valid && autoCreate) {
+                valid = autoCreateEPersonMetadataField(context, metadataFieldName);
+            }
+
+            if (valid) {
+                // The eperson field is fine, we can use it.
+
+                log.debug("Loading additional eperson metadata mapping for: {}={}",
+                        attributeName, metadataFieldName);
+
+                map.put(attributeName, metadataFieldName);
+            } else {
+                // The field doesn't exist, and we can't use it.
+
+                log.error("Skipping the additional eperson metadata mapping for: {}={}"
+                        + " because the field is not supported by the current configuration.",
+                        attributeName, metadataFieldName);
+            }
+        }
+
+        metadataHeaderMap = map;
+    }
+
+    /**
+     * Check if a metadata field for an EPerson is available.
+     *
+     * @param metadataName The name of the metadata field.
+     * @param context      context
+     * @return True if a valid metadata field, otherwise false.
+     * @throws SQLException if database error
+     */
+    protected synchronized boolean checkIfEpersonMetadataFieldExists(Context context, String metadataName)
```

Comment:
The camelCase on this method name is a bit inconsistent compared to `registerNewEPerson` and EPerson class name etc but I see that's also the case with `ShibAuthentication` so, not a blocker or requirement to fix imo

Reply:
Fixed in 8f4aacd.